syntax = "proto3";

message PubSubJobMessage {
    string jobId = 1;
    string jobType = 2;
    oneof jobArgs {
        VideoTranscodeJobArgs videoTranscodeJobArgs = 3;

    }
}

message VideoTranscodeJobArgs {
    string bucketName = 1;
    string gsRawVideo = 2;
    string format = 3;
    string convertedVideoBucket = 4;
    string convertedVidePath = 5;
}

/*

payload = {
            "job_id": "{}".format(uuid.uuid4()),
            "job_type": "video-transcode",
            "job_args": {
                "bucket_name": namespace.bucket_name,
                "gs_raw_video": namespace.gs_raw_video,
                "format": namespace.format,
                "output_sizes": namespace.output_sizes,
                "converted_video_bucket": namespace.converted_video_bucket,
                "converted_video_path": namespace.converted_video_path,
            }
        }

*/